### PR TITLE
[message-buffer 15/x] - audit fixes

### PR DESCRIPTION
**Pull request by swimricky**
### Summary
Address fixes from TOB audit report

### Changes
1. Initialize - remove admin pubkey parameter & rename `payer` to `admin` in `Initialize` context and use admin.key() as `whitelist.admin` (TOB-PYTH-5)
2. Delete - remove auth check (TOB-PYTH-7)
3. Resize - add explicit max size of `u16::MAX + MessageBuffer::HEADER_LEN` and add check (TOB-PYTH-10)
4. PutAll 
    1. add bounds check for end_offset index (TOB-PYTH-9)
    2. fix truncation in `put_all_in_buffer` (TOB-PYTH-10)
5. use consts for seeds

### CODE REVIEW

File: pythnet/message_buffer/programs/message_buffer/src/lib.rs [hunk 1]
```diff
@@ -18,7 +21,8 @@ pub mod message_buffer {
 
     /// Initializes the whitelist and sets it's admin to the provided pubkey
     /// Once initialized, the authority must sign all further changes to the whitelist.
-    pub fn initialize(ctx: Context<Initialize>, admin: Pubkey) -> Result<()> {
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        let admin = ctx.accounts.admin.key();
```
Comment (guibescos) [inline pythnet/message_buffer/programs/message_buffer/src/lib.rs:25]: I think `whitelist.admin = ctx.accounts.admin.key();` is easier to read

File: pythnet/message_buffer/programs/message_buffer/src/instructions/resize_buffer.rs [hunk 2]
```diff
@@ -47,15 +55,16 @@ pub struct ResizeBuffer<'info> {
     pub system_program: Program<'info, System>,
```
Comment (guibescos) [inline pythnet/message_buffer/programs/message_buffer/src/instructions/resize_buffer.rs:55]: I think you missed some places in replacing "message" by "MESSAGE"

**Review by ali-bahjati — APPROVED** [review]
Nice!

File: pythnet/message_buffer/programs/message_buffer/src/lib.rs (hunk 1 above)
Comment (swimricky) [reply pythnet/message_buffer/programs/message_buffer/src/lib.rs]: done thanks

File: pythnet/message_buffer/programs/message_buffer/src/instructions/resize_buffer.rs (hunk 2 above)
Comment (swimricky) [reply pythnet/message_buffer/programs/message_buffer/src/instructions/resize_buffer.rs:56]: fixed thanks

**Pull request merged**